xpk.py

```diff
@@ -2490,17 +2783,49 @@ def workload_create(args) -> int:
   else:
     container = get_main_container(args, system, docker_image, resource_type)
 
-  yml_string = workload_create_yaml.format(args=args,
-                                           system=system,
-                                           container=container,
-                                           affinity=get_cpu_affinity(system.accelerator_type),
-                                           env=get_cpu_env(args.num_slices,system),
-                                           accelerator_label=create_accelerator_label(system.accelerator_type, system),
-                                           machine_label=create_machine_label(system.accelerator_type, system),
-                                           local_queue_name=_LOCAL_QUEUE_NAME)
-  tmp = write_temporary_file(yml_string)
-  command = f'kubectl apply -f {str(tmp.file.name)}'
-  return_code = run_command_with_updates(command, 'Creating Workload', args)
+  if args.use_pathways:
+    # Ensure the cluster and CPU nodepools were created with --enable-pathways
+    all_node_pools = get_all_nodepools_programmatic(args)
+    desired_pw_cpu_node_pools = {'cpu-user-np', 'cpu-rm-np', 'cpu-proxy-np'}
+    if not desired_pw_cpu_node_pools.issubset(set(all_node_pools[0])):
+      xpk_print(
+          'Cluster needs to be created with --enable-pathways to run Pathways workloads.'
```

Comment:
nit Could add that the reason this errored is because there are missing cpu node pools, and print the missing ones?

Reply:
Sure, makes sense